src/PIL/DdsImagePlugin.py

```diff
@@ -148,15 +148,17 @@ def _open(self):
             self.tile = [("raw", (0, 0) + self.size, 0, (self.mode, 0, 1))]
         elif pfflags & DDPF_RGB:
             # Texture contains uncompressed RGB data
-            masks = {mask: ["R", "G", "B", "A"][i] for i, mask in enumerate(masks)}
-            rawmode = ""
-            if pfflags & DDPF_ALPHAPIXELS:
-                rawmode += masks[0xFF000000]
-            else:
+            if not pfflags & DDPF_ALPHAPIXELS:
                 self._mode = "RGB"
-            rawmode += masks[0xFF0000] + masks[0xFF00] + masks[0xFF]
+            if masks[:3] == (31744, 992, 31) and self.mode == "RGB":
+                rawmode = "BGR;15"
+            else:
+                masks = {mask: ["R", "G", "B", "A"][i] for i, mask in enumerate(masks)}
```

Comment:
```suggestion
                masks = {mask: "RGBA"[i] for i, mask in enumerate(masks)}
```
would be more efficient, since the (possibly-mutable-but-the-interpreter-can't-tell) list doesn't need to regenerated for every call of this function, but it could use the same `"RGBA"` string constant used just below.

Reply:
I've added a version of this.